torch/csrc/inductor/aoti_eager/kernel_holder.cpp

```diff
@@ -0,0 +1,283 @@
+  if (kernel_lib_path.empty()) {
+    TORCH_CHECK(
+        has_fall_back_,
+        "Failed to produce kernel libarary by using AOTI and no fall back");
+    python_kernel_holder_(op, keyset, stack);
+    return;
+  }
+
+  auto device_index = 0; // TODO: Get device index from other tensors.
+  auto device = c10::Device(device_type, device_index);
+
+  std::shared_ptr<AOTIModelContainerRunner> kernel = nullptr;
+  if (device_type == c10::DeviceType::CPU) {
+    kernel = std::make_shared<AOTIModelContainerRunnerCpu>(kernel_lib_path);
+  } else {
+#ifdef USE_CUDA
+    kernel = std::make_shared<AOTIModelContainerRunnerCuda>(kernel_lib_path);
+#else
+    TORCH_CHECK(false, "Unsupported CUDA device type");
+#endif
+  }
+
+  std::vector<at::Tensor> inputs;
+  if (unpackTensors(*stack, device, inputs)) {
+    auto outputs = kernel->run(inputs);
+    if (outputs.size() > 0) {
+      torch::jit::drop(*stack, op.schema().arguments().size());
+      // TODO: Get the output type of this operation and then convert to the
+      // output type.
+      for (auto& output : outputs) {
+        torch::jit::push(*stack, std::move(output));
+      }
+    }
+  }
+}
+
+std::string AOTIPythonKernelHolder::produce_aoti_kernel_lib(
+    const c10::OperatorHandle& op,
+    c10::DispatchKeySet keyset,
+    torch::jit::Stack* stack) {
+  auto arguments = torch::jit::last(*stack, op.schema().arguments().size());
+  std::string kernel_lib_path("");
+
+  py::gil_scoped_acquire gil;
+
+  // Get the corresponding python operation for the current operator and the
+  // python operation will pass to the AOT Inductor to generate the kernel
+  // library.
+  const auto& schema = op.schema();
+  const auto& qualified_name = op.operator_name().name;
+  const auto& overload_name =
+      schema.overload_name().empty() ? "default" : schema.overload_name();
+  auto pos = qualified_name.find("::");
+  TORCH_INTERNAL_ASSERT(pos != std::string::npos, qualified_name);
+  // Make me some null terminated strings
+  std::string ns_str = qualified_name.substr(0, pos);
+  const char* ns = ns_str.c_str();
+  const char* func_name = qualified_name.c_str() + pos + strlen("::");
+  py::handle op_py_func = op.getPythonOp(pyinterpreter_, [&]() -> PyObject* {
+    py::handle torch_api_function =
+        py::module::import("torch").attr("ops").attr(ns).attr(func_name);
+    return torch_api_function.attr(overload_name.c_str()).ptr();
+  });
+
+  if (op_py_func.ptr() == nullptr || op_py_func.ptr() == Py_None) {
+    TORCH_WARN(
+        "Failed to get python operation. Operator Name is ",
+        op.operator_name().name,
+        ", Overload Name is ",
+        overload_name);
+    return kernel_lib_path;
+  }
+
+  py::handle aot_compile_function =
+      py::module::import("torch._export").attr("aot_compile");
+  if (aot_compile_function.ptr() == nullptr ||
+      aot_compile_function.ptr() == Py_None) {
+    TORCH_WARN("Failed to import - torch._export.aot_compile");
+    return kernel_lib_path;
```

Comment:
Should this be an error, when does it happen?

Reply:
Thanks. I replaced it with assert.